packages/portal-proto/src/components/InputEntityList/MatchedTable.tsx

```diff
@@ -92,13 +82,34 @@ const MatchedTable = ({
               meta: {
                 sortingFn: (rowA, rowB) => {
                   const property = `submitted_${id.replaceAll(".", "_")}`;
-                  if (rowA[property] > rowB[property]) {
-                    return 1;
+
+                  const valueA = rowA[property];
+                  const valueB = rowB[property];
+
+                  // if values are undefined i.e., "--"
+                  if (!valueA && !valueB) return 0;
+                  if (!valueA) return 1;
+                  if (!valueB) return -1;
+
+                  // check for HGNC
+                  const isHGNC = (value: string) => value.startsWith("HGNC:");
```

Comment:
Minor performance thing, can you define this function outside of the component?

Reply:
just updated it